piptools/exceptions.py

```diff
@@ -6,12 +11,17 @@ class PipToolsError(Exception):
 
 
 class NoCandidateFound(PipToolsError):
-    def __init__(self, ireq, candidates_tried, finder):
+    def __init__(
+        self,
+        ireq: InstallRequirement,
+        candidates_tried: List[InstallationCandidate],
```

Comment:
```suggestion
        candidates_tried: Iterable[InstallationCandidate],
```

Could be relaxed to pass any iterables

Reply:
Thanks! I applied this suggestion in the latest revision.